src/main/java/com/example/java/maven/game2048/Board.java

```diff
@@ -18,18 +23,258 @@ public Board(int numberOfRows, int numberOfColumns) {
+            sumInRowOrColumnLeftOrUp(row);
+        }
+    }
+
+    public void sumDown() {
+        for (Field[] column : getColumns()) {
+            sumInRowOrColumnRightOrDown(column);
+        }
+    }
+
+    public void sumUp() {
+        for (Field[] column : getColumns()) {
+            sumInRowOrColumnLeftOrUp(column);
+        }
+    }
+
+    public void sumRight() {
+        for (Field[] row : this.fields) {
+            sumInRowOrColumnRightOrDown(row);
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        int result = Objects.hash(numberOfRows, numberOfColumns);
+        result = 31 * result + Arrays.hashCode(fields);
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Board board = (Board) o;
+        return numberOfRows == board.numberOfRows &&
+                numberOfColumns == board.numberOfColumns &&
+                Arrays.deepEquals(fields, board.fields);
+    }
+
     @Override
     public String toString() {
         StringBuilder strBoardBuilder = new StringBuilder();
         for (Field[] row : this.fields) {
             strBoardBuilder.append(
                     Arrays.toString(row)
-                            .replace(',', ' ')
-                            .replace('[', ' ')
-                            .replace(']', ' ')
+                            .replace(",", "")
+                            .replace("[", "")
+                            .replace("]", "")
             );
             strBoardBuilder.append("\n");
         }
-        return strBoardBuilder.toString();
+        return strBoardBuilder.toString().trim();
+    }
+
+    public boolean hasWinningNumber(int winningNumber) {
+        for (Field[] row : this.fields) {
+            for (Field field : row) {
+                if (field.getValue() == winningNumber) {
+                    return true;
+                }
+            }
+        }
+        return false;
     }
+
+    public boolean hasSimilarNeighboringFields() {
+        for (Field[] row : this.fields) {
+            for (int index = 0; index < row.length - 1; index++) {
+                int currentFieldValue = row[index].getValue();
+                int nextFieldValue = row[index + 1].getValue();
+                if (currentFieldValue == nextFieldValue) {
+                    return true;
+                }
+            }
+        }
+        for (Field[] column : getColumns()) {
+            for (int index = 0; index < column.length - 1; index++) {
+                int currentFieldValue = column[index].getValue();
+                int nextFieldValue = column[index + 1].getValue();
+                if (currentFieldValue == nextFieldValue) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public List<Field> getFieldsWithZeroValue() {
```

Comment:
You should not return `Field` anywhere outside of `Board`. It gives possibility to modify `fields` in `board` object outside of it. So for example now I can in game make sth like:
```java
List<Field> fields = board.getFieldsWithZeroValue();
fields.forEach(field -> field.setValue(2048);
```
I know that from player perspective nothing changed. But from the design perspective it is dangerous, especially in projects which are big. Someone new in project can change behavior of your program by mistake. This is a reason why in Java we have encapsulation (public/protected/private and package private).